### PR TITLE
Sdk Implementation for message center

### PR DESCRIPTION
**Pull request by sobhanagit**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/SpotinstRepoManager.java [hunk 1]
```diff
@@ -46,6 +47,7 @@ class SpotinstRepoManager {
     private ISpotAzureStatefulNodeRepo                  spotAzureStatefulNodeRepo;
     private ISpotAzureAksVngRepo    					          spotAksVngRepo;
     private ISpotAdminOrganizationRepo                  spotAdminOrganizationRepo;
+    private IMessageCenterRepo                          spotInstMessageCenterRepo;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstRepoManager.java:50]: Change spotInstMessageCenterRepo to messageCenterRepo

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiBulkUpdateMessages.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.spotinst.sdkjava.model.api.admin.messageCenter;
+
+import com.fasterxml.jackson.annotation.*;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+import com.spotinst.sdkjava.model.bl.admin.messageCenter.Entities;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiBulkUpdateMessages.java:5]: should import api.admin.messageCenter.apiEntities, not this one.

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiCreateMessage implements IPartialUpdateEntity {
+    //region Members
+    @JsonIgnore
+    private Set<String> isSet;
+    private String title;
+    private String  content;
+    private String stickyUntil;
+    private String type;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:21]: Can we define type as enumeration?

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+    private String mainIcon;
+    private String hyperlink;
+    private String secondaryIcon;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+    private Date timestamp;
+    private String audienceSpecification;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:27]: Should be defined as enumeration?

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiCreateMessage implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:14]: Use different object for Get response.

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java [hunk 6]
```diff
@@ -0,0 +1,282 @@
+package com.spotinst.sdkjava.model.api.admin.messageCenter;
+
+import com.fasterxml.jackson.annotation.*;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+import com.spotinst.sdkjava.model.bl.admin.messageCenter.Metadata;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java:5]: Why bl class is included. This is api side?

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiGetMessage implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java:15]: Is this response class for the Get message?

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/CreateMessage.java [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CreateMessage {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/CreateMessage.java:15]: Same comment as API class.

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class GetMessage {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java:15]: If this a response object rename it to GetMessageResponse

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+    private String title;
+    private String content;
+    private String stickyUntil;
+    private String type;
+    private String mainIcon;
+    private String hyperlink;
+    private String secondaryIcon;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+    private Date timestamp;
+    private String audienceSpecification;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java:28]: Confirm if these 3 fields are needed in the response object.

File: src/main/java/com/spotinst/sdkjava/model/converters/admin/messageCenter/MessageCenterConverter.java [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+            if (userOrganizationMapping.isUserIdsSet()) {
+                apiUsersAudienceSpecification.setUserIds(userOrganizationMapping.getUserIds());
+            }
+
+        }
+
+        return apiUsersAudienceSpecification;
+    }
+
+    public static GetMessage toBl(ApiGetMessage apiGetMessage) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/admin/messageCenter/MessageCenterConverter.java:146]: All fields defined in GetMessage class are not converted here. Please confirm which is the correct set of parameters in this object.

File: src/main/java/com/spotinst/sdkjava/model/repo/admin/messageCenter/MessageCenterRepo.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+        }
+        catch (SpotinstHttpException e) {
+            retVal = ExceptionHelper.handleHttpException(e);
+        }
+        return retVal;
+    }
+
+    @Override
+    public RepoGenericResponse<Boolean> bulkUpdate(BulkUpdateMessages messageToUpdate, String authToken) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/repo/admin/messageCenter/MessageCenterRepo.java:66]: rename variable to messagesToUpdate

File: src/main/java/com/spotinst/sdkjava/model/requests/admin/account/messageCenter/ListAllMessagesRequest.java [hunk 13]
```diff
@@ -0,0 +1,88 @@
+package com.spotinst.sdkjava.model.requests.admin.account.messageCenter;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/requests/admin/account/messageCenter/ListAllMessagesRequest.java:1]: Incorrect package. Should be in requests.admin.messageCenter.

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Review completed. Please check the review comments and fix.

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiBulkUpdateMessages.java (hunk 2 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiBulkUpdateMessages.java:5]: It is an unused import removed now

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java (hunk 4 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:27]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java (hunk 6 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java:5]: Unused import deleted

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java (hunk 7 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiGetMessage.java:15]: Yes

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/CreateMessage.java (hunk 8 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/CreateMessage.java:15]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java (hunk 9 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java:15]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java (hunk 10 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/bl/admin/messageCenter/GetMessage.java:28]: Not used Removed

File: src/main/java/com/spotinst/sdkjava/model/converters/admin/messageCenter/MessageCenterConverter.java (hunk 11 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/converters/admin/messageCenter/MessageCenterConverter.java:146]: Removed unused parameters

File: src/main/java/com/spotinst/sdkjava/model/repo/admin/messageCenter/MessageCenterRepo.java (hunk 12 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/repo/admin/messageCenter/MessageCenterRepo.java:66]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/requests/admin/account/messageCenter/ListAllMessagesRequest.java (hunk 13 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/requests/admin/account/messageCenter/ListAllMessagesRequest.java:1]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java (hunk 3 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:21]: Resolved

File: src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java (hunk 5 above)
Comment (sobhanagit) [reply src/main/java/com/spotinst/sdkjava/model/api/admin/messageCenter/ApiCreateMessage.java:14]: Resolved

**Comment by anuragsharma-123** [comment]
This PR is no longer valid.

**Pull request closed without merging**